effectful/handlers/pyro.py

```diff
@@ -302,15 +303,8 @@ def named_distribution(
     *names: Annotated[Operation[[], torch.Tensor], Scoped[B]],
 ) -> Annotated[TorchDistribution, Scoped[A | B]]:
     d = defterm(d)
-    dist_constr, args = d.args[0], d.args[1:]
 
-    if not (
-        d.op is call
-        and (
-            issubclass(dist_constr, TorchDistribution)
-            or issubclass(dist_constr, dist.torch_distribution.TorchDistributionMixin)
-        )
-    ):
+    if not isinstance(d, _DistributionTerm):
```

Comment:
As mentioned in another comment, semantic functions should not depend on `Term` subtypes that are implementation details of `defterm`/`defdata`.